src/pick.c

```diff
@@ -446,6 +448,14 @@ selected_choice(void)
 			while (cursor_position < query_length
 			    && isu8cont(query[++cursor_position]));
 			break;
+		case PAGE_DOWN:
+			if (yscroll + lines - 1 < choices_count)
```

Comment:
How does this handle the case where there are still choices below the selection and possibly below the scroll, but not more than `lines - 1`.

If a full page can't be scrolled (if this condition is false), I'd expect it to scroll to and select the last choice. What are your thoughts?